### PR TITLE
Add new users

### PR DESCRIPTION
**Pull request by KjartanE**
* Added "+ new users" button on Admin user page.

* formatted text with prettier, deleted test comment

* package lock update

# Overview

## Links to jira tickets

- https://quartech.atlassian.net/browse/BHBC-1520?atlOrigin=eyJpIjoiYmMxM2QwZDM5OGFlNDFhY2E2YzU1YTY3ZTE5Mzk3YWUiLCJwIjoiaiJ9
- not directly part of this ticket but an addition to it.

## This PR contains the following changes

- Under the "Manage User" page (/admin/users) added a "+ New Users" button to allow system user to add a new active user and assign them a role.

## This PR contains the following types of changes

- [ x ] New feature (change which adds functionality)
- [ ] Enhancement (improvements to existing functionality)
- [ ] Bug fix (change which fixes an issue)
- [ ] Misc cleanup / Refactoring / Documentation
- [ ] Version change

## Checklist

A list of items that are good to consider when making any changes.

_Note: this list is not exhaustive, and not all items are always applicable._

### General

- [x] I have performed a self-review of my own code

### Code

- [ x ] New files/classes/functions have appropriately descriptive names and comment blocks to describe their use/behaviour
- [ ] I have avoided duplicating code when possible, moving re-usable pieces into functions
- [ ] I have avoided hard-coding values where possible and moved any re-usable constants to a constants file
- [ ] My code is as flat as possible (avoids deeply nested if/else blocks, promise chains, etc)
- [ ] My code changes account for null/undefined values and handle errors appropriately
- [ ] My code uses types/interfaces to help describe values/parameters/etc, help ensure type safety, and improve readability

### Style

- [ x ] My code follows the established style conventions
- [ x ] My code uses native material-ui components/icons/conventions when possible

### Documentation

- [ ] I have commented my code sufficiently, such that an unfamiliar developer could understand my code
- [ ] I have added/updated README's and related documentation, as needed

### Tests

- [ ] I have added/updated unit tests for any code I've added/updated
- [ ] I have added/updated the Postman requests/tests to account for any API endpoints I've added/updated

### Linting/Formatting

- [x] I have run the linter and fixed any issues, as needed  
       _See the `lint` commands in package.json_
- [x] I have run the formatter and fixed any issues, as needed  
       _See the `format` commands in package.json_

### SonarCloud

- [x] I have addressed all SonarCloud Bugs, Vulnerabilities, Security Hotspots, and Code Smells

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Screenshots

Please add any relevant UI screenshots, if applicable.

### CODE REVIEW

File: api/src/paths/user/add.ts
Comment (anissa-agahchen) [inline api/src/paths/user/add.ts]: suggestion: move this down to line 115, just before you use it

File: app/src/features/admin/users/ActiveUsersList.test.tsx
Comment (anissa-agahchen) [inline app/src/features/admin/users/ActiveUsersList.test.tsx]: can you add an assertion that involves the test-id? It's more robust than looking for text

File: app/src/features/admin/users/ActiveUsersList.tsx
Comment (NickPhura) [inline app/src/features/admin/users/ActiveUsersList.tsx]: Maybe double check with Jeremy what the wording here should be. "Team Members" is applicable at the project level, but probably not at the system level.

File: app/src/features/admin/users/ActiveUsersList.tsx
Comment (NickPhura) [inline app/src/features/admin/users/ActiveUsersList.tsx]: Getting a small warning about `error` being unknown.  Need to cast `error` to `APIError`. 

In general, (regarding `dialogError` and `dialogErrorDetails`) they aren't set consistently everywhere, but ideally they should both be set as below:

```typescript
      openErrorDialog({
        ...defaultErrorDialogProps,
        open: true,
        dialogError: (error as APIError).message,
        dialogErrorDetails: (error as APIError).errors
      });
```

Comment (NickPhura) [reply app/src/features/admin/users/ActiveUsersList.tsx]: There are some tech debt improvements on the to-do list to audit the error handling in general.  And ensure the API returns sensible error message, etc.

File: app/src/features/admin/users/ActiveUsersList.tsx
Comment (NickPhura) [inline app/src/features/admin/users/ActiveUsersList.tsx:352]: Something I didn't account for in the `ProjectParticipantsHeader` component:

If the save function fails, we still render the "X users added" message.
Easy fix is to move the `dialogContext.setSnackbar({...})` call into the save function, so that if an error is thrown, it gets skipped. I would just put it after the `props.refresh()` call.

Feel free to make the same change in `ProjectParticipantsHeader` as well, if you please :)

File: app/src/features/admin/users/ActiveUsersList.test.tsx
Comment (NickPhura) [inline app/src/features/admin/users/ActiveUsersList.test.tsx]: I'd be more specific: `renders the add new users button correctly`, or something like that

File: app/src/features/admin/users/AddSystemUsersForm.tsx
Comment (NickPhura) [inline app/src/features/admin/users/AddSystemUsersForm.tsx]: Maybe a bit of an annoying change, but can you change the `participants`/`participant` language to something like `users` or `systemUsers`.  "Participants" is something that only applies to Projects (in the database).  The code isn't wrong, but it will be slightly nicer to read for a future dev.

**Review by NickPhura — APPROVED** [review]
Looks good.  When you click the merge button, just make sure to unify the commit messages into a single message that sums up the changes (just something brief). I would also prefix the commit name to include the jira ticket number: `BHBC-1520: Add new users (#675)` or something similar.  Can also click the "Delete Branch" button after merging too, to keep our old branches in check.

**Pull request merged**
